sound/soc/sof/ipc4-pcm.c

```diff
@@ -294,6 +296,15 @@ static int sof_ipc4_chain_dma_trigger(struct snd_sof_dev *sdev,
 			msg.extension |= pipeline->msg.extension;
 	}
 
+	if (direction == SNDRV_PCM_STREAM_CAPTURE) {
+		/*
+		 * for capture we need to add num_playback to the value (stream_tag - 1)
+		 * stored in msg.primary
+		 */
+		msg.primary +=  SOF_IPC4_GLB_CHAIN_DMA_HOST_ID(ipc4_data->num_playback_streams);
+		msg.primary +=  SOF_IPC4_GLB_CHAIN_DMA_LINK_ID(ipc4_data->num_playback_streams);
```

Comment:
Do we need to check if dma host id and dma link id exceeds the max id after adding num_playback?

Reply:
> Do we need to check if dma host id and dma link id exceeds the max id after adding num_playback?

@bardliao, I would expect that we already done checks when the stream_tag was picked?

Reply:
we could be paranoid indeed, but there is already limits on stream_tag already. I am with @ujfalusi it's probably fine.

Reply:
no change.